src/components/Thread.vue

```diff
@@ -100,12 +104,28 @@ export default {
 				this.expandedThreads = this.expandedThreads.filter(t => t !== threadId)
 			}
 		},
+		onMove(threadId) {
+			if (threadId === this.threadId) {
+				this.$router.replace({
+					name: 'mailbox',
+					params: {
+						mailboxId: this.$route.params.mailboxId,
+					},
+				})
+			} else {
+				this.expandedThreads = this.expandedThreads.filter((id) => id !== threadId)
+				this.fetchThread()
+			}
+		},
+		resetThread() {
```

Comment:
can we make this an async fn and await the fetch? Even though we'll call it sync (and ignore the returned promise) right now I think it should be an awaitable fn.

Reply:
Sure